components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/DefaultServiceURLBuilder.java

```diff
@@ -0,0 +1,426 @@
+             */
+            @Override
+            public String getHostName() {
+
+                return hostName;
+            }
+
+            /**
+             * This method is called to get the port of Service URL.
+             *
+             * @return value of the port.
+             */
+            @Override
+            public int getPort() {
+
+                return port;
+            }
+
+            /**
+             * This method is called to get the Url path of Service URL.
+             *
+             * @return String of the url path.
+             */
+            @Override
+            public String getUrlPath() {
+
+                return resolvedUrlStringBuilder.toString();
+            }
+
+            /**
+             * This method is called to get the parameter value when the key is provided.
+             *
+             * @param key Key of the parameter.
+             * @return The value of the parameter.
+             */
+            @Override
+            public String getParameter(String key) {
+
+                return parameters.get(key);
+            }
+
+            /**
+             * This method is called to get a map of the parameters.
+             *
+             * @return The parameters.
+             */
+            @Override
+            public Map<String, String> getParameters() {
+
+                return Collections.unmodifiableMap(parameters);
+            }
+
+            /**
+             * This method is called to get a list of the parameter names.
+             *
+             * @return List of parameter keys.
+             */
+            @Override
+            public List<String> getParameterKeys() {
+
+                ArrayList<String> parameterArrayList = new ArrayList<>();
+
+                for (Map.Entry<String, String> entry : parameters.entrySet()) {
+                    parameterArrayList.add(entry.getKey());
+                }
+                return Collections.unmodifiableList(parameterArrayList);
+            }
+
+            /**
+             * This method is called to get the decoded fragment from the url.
+             *
+             * @return The decoded fragment.
+             */
+            @Override
+            public String getFragment() {
+
+                return resolvedFragment;
+            }
+
+            /**
+             * This method is called to get the tenant domain appended in the URL.
+             *
+             * @return The tenant domain.
+             */
+            @Override
+            public String getTenantDomain() {
+
+                return tenantDomain;
+            }
+
+            /**
+             * Concatenate the protocol, host name, port, proxy context path, web context root and url context to
+             * return the absolute URL.
+             *
+             * @return The absolute URL from the Service URL instance.
+             */
+            @Override
+            public String getAbsoluteURL() throws URLBuilderException {
```

Comment:
It's fine for the builder to throw URLBuilderException while building the URL. But ServiceURL POJO should not throw it when getting absolute URL or relative URL